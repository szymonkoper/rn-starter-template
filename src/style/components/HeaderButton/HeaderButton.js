import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/native'
import { fontSizeS, screenMargin } from '../../dimensions'
import Text from '../Text'

const TouchableContainer = styled.TouchableOpacity`
  margin-horizontal: ${screenMargin}px;
`

const TextContent = styled(Text)`
  font-size: ${fontSizeS};
`

const HeaderButton = ({ onPress, title }) => (
  <TouchableContainer onPress={onPress}>
    <TextContent>{title}</TextContent>
  </TouchableContainer>
)

HeaderButton.propTypes = {
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired
}

export default HeaderButton
