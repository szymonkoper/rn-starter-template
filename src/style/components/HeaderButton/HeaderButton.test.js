import React from 'react'
import { fireEvent, render } from 'react-native-testing-library'
import HeaderButton from './HeaderButton'

const BUTTON_TITLE = "A header button's title"

describe('HeaderButton', () => {
  let props = null

  const renderComponent = () =>
    // eslint-disable-next-line react/jsx-props-no-spreading
    render(<HeaderButton {...props} />)

  beforeEach(() => {
    props = {
      onPress: jest.fn(),
      title: BUTTON_TITLE
    }
  })

  it('renders correctly', () => {
    const { toJSON } = renderComponent(props)
    expect(toJSON()).toMatchSnapshot()
  })

  it('calls onPress correctly', () => {
    const { getByText } = renderComponent(props)

    fireEvent.press(getByText(BUTTON_TITLE))

    expect(props.onPress).toHaveBeenCalledTimes(1)
  })
})
