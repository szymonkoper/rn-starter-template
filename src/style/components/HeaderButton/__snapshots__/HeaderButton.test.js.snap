// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginHorizontal": 16,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 18,
        },
      ]
    }
  >
    A header button's title
  </Text>
</View>
`;
